### PR TITLE
contribution

### PR DESCRIPTION
**Pull request by BilalAlam173**
1-login page link created on "join us" button at navigation
2-signup page link created on login page 
3-redirect to login page on successfull signup
4-redirect to profile page on successfull login and displays logout option and user name at navigation else display error

### CODE REVIEW

**Comment by BilalAlam173** [comment]
@mashhoodr ready for review

File: src/app/auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
+      if(localStorage.getItem('isLoggedin') === 'true') {
```
Comment (mashhoodr) [inline src/app/auth.guard.ts]: @BilalAlam173 please maintain the user state in user.service and fetch it from local storage once in the beginning

File: src/app/login/login.component.ts [hunk 2]
```diff
@@ -11,21 +12,25 @@ export class LoginComponent implements OnInit {
...
-    // this.UserService.login(this.email,this.password).then(function(response){
-    //   console.log(response);
-    // });
-    this.UserService.login(this.email,this.password).subscribe( data => {
-      console.log(data);
+    this.UserService.login(this.email, this.password).subscribe( data => {
+      console.log(data.data);
+      if(data.data) {
+      localStorage.setItem('isLoggedin', 'true');
+      localStorage.setItem('user', JSON.stringify(data.data));
```
Comment (mashhoodr) [inline src/app/login/login.component.ts]: @BilalAlam173 arent we using JSON Webtokens for authentication? We need that for authentication

File: src/app/login/login.component.ts [hunk 3]
```diff
@@ -11,21 +12,25 @@ export class LoginComponent implements OnInit {
...
+      window.location.reload();
```
Comment (mashhoodr) [inline src/app/login/login.component.ts]: We should not need to reload.

File: src/app/shared/navigation/navigation.component.ts [hunk 4]
```diff
@@ -7,9 +8,29 @@ import { Component, OnInit } from '@angular/core';
...
+  showOptions() {
+    this.nameHover = 'btn-trial';
+  }
+  hideOptions(){
+    this.nameHover = '';
+  }
 
   ngOnInit() {
+    if(localStorage.getItem('isLoggedin')){
+      this.userName = JSON.parse(localStorage.getItem('user')).name;
```
Comment (mashhoodr) [inline src/app/shared/navigation/navigation.component.ts]: maintain the user as a variable in the service.

**Comment by mashhoodr** [comment]
@BilalAlam173 any updates on this?

File: server/routes/user.route.js [hunk 5]
```diff
@@ -78,21 +79,40 @@ router.post('/', function(req, res, next) {
...
+        
+        throw err;
+        
+      }else{
+        if(user){
+          const payload = {
+            user: user 
+          };
+          
+          var token = jwt.sign(payload,'superSecret', {
```
Comment (afirdousi) [inline server/routes/user.route.js]: Shouldn't this `superSecret` go as node env variable and not mentioned here explicitly?

File: src/app/services/user.service.ts [hunk 6]
```diff
@@ -1,41 +1,54 @@
...
-
 import { user } from '../model/user.interface';
 
 @Injectable()
 export class UserService {
+
   private api = '/api/v1/user';
 
-  constructor(private http: Http,private ErrorService) { }
+  constructor(private http: Http, private ErrorService: ErrorService) {}
```
Comment (afirdousi) [inline src/app/services/user.service.ts]: the param name "ErrorService" needs to be camel cased to **errorService**

Comment (afirdousi) [reply src/app/services/user.service.ts]: also please replace all references after that. I hope your IDE will.

File: src/app/shared/footer/footer.component.html [hunk 7]
```diff
@@ -4,25 +4,25 @@
 
     <h3>Join Us</h3>
 
-    <form class="mc-trial row">
+    <!-- <form class="mc-trial row">
```
Comment (afirdousi) [inline src/app/shared/footer/footer.component.html]: @mashhoodr this quick `join us` used to work. So are we taking it off completely in favor of the sign-up module?

**Pull request merged**